### PR TITLE
chore: split api-platform/elasticsearch

### PR DESCRIPTION
**Pull request by Zowac**
| Q             | A
| ------------- | ---
| Branch?       | 3.1 
| License       | MIT

### CODE REVIEW

File: src/Elasticsearch/Exception/IndexNotFoundException.php [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 namespace ApiPlatform\Elasticsearch\Exception;
 
-use ApiPlatform\Exception\ExceptionInterface;
+use ApiPlatform\Metadata\Exception\ExceptionInterface;
```
Comment (soyuka) [inline src/Elasticsearch/Exception/IndexNotFoundException.php:16]: not sure about this

File: src/Metadata/IriConverterInterface.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+use ApiPlatform\Metadata\Exception\InvalidArgumentException;
+use ApiPlatform\Metadata\Exception\ItemNotFoundException;
+use ApiPlatform\Metadata\Exception\RuntimeException;
+
+/**
+ * Converts item and resources to IRI and vice versa.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+interface IriConverterInterface
```
Comment (soyuka) [inline src/Metadata/IriConverterInterface.php:25]: not sure about this

File: src/Metadata/UrlGeneratorInterface.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ * We are using the term "URL" instead of "URI" as this is more common in web applications
+ * and we do not need to distinguish them as the difference is mostly semantical and
+ * less technical. Generating URIs, i.e. representation-independent resource identifiers,
+ * is also possible.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Tobias Schultze <http://tobion.de>
+ * @copyright Fabien Potencier
+ */
+interface UrlGeneratorInterface
```
Comment (soyuka) [inline src/Metadata/UrlGeneratorInterface.php:36]: not sure about this

File: src/Elasticsearch/Exception/IndexNotFoundException.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (soyuka) [reply src/Elasticsearch/Exception/IndexNotFoundException.php:16]: If Elasticsearch needs an ExceptionInterface, add one it the `Elasticsearch` namespace.

File: src/Api/IriConverterInterface.php [hunk 4]
```diff
@@ -13,33 +13,11 @@
 
 namespace ApiPlatform\Api;
 
-use ApiPlatform\Exception\InvalidArgumentException;
-use ApiPlatform\Exception\ItemNotFoundException;
-use ApiPlatform\Exception\RuntimeException;
-use ApiPlatform\Metadata\Operation;
+interface_exists(\ApiPlatform\Metadata\IriConverterInterface::class);
```
Comment (PierreRebeilleau) [inline src/Api/IriConverterInterface.php:16]: `interface IriConverterInterface  extends \ApiPlatform\Metadata\IriConverterInterface`

**Review by soyuka — APPROVED** [review]

**Pull request merged**